starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,108 @@
 // Names and Input
-
+//1.
+var hacker1 = "Juan";
+//2.
+console.log("The driver's name is " + hacker1);
+//3.
+var hacker2 = prompt("What's the navigator's name?");
+//4.
+console.log("The navigator's name is " + hacker2);
 
 //Conditionals
+//5.
+if (hacker1.length > hacker2.length) {
+  console.log(
+    "The driver has the longest name, it has " + hacker1.length + " characters"
+  );
+} else if (hacker1.length < hacker2.length) {
+  console.log(
+    "Yo, the navigator got the longest name, it has " +
+      hacker2.length +
+      " characters"
+  );
+} else {
+  console.log(
+    "Wow, you both got equally long names, " + hacker1 + " characters!!"
+  );
+}
+
+//Loops
+//6.
+var result = "";
+for (let i = 0; i < hacker1.length; i++) {
+  result += hacker1.toUpperCase()[i] + " ";
+}
+console.log(result);
+
+//7.
+var result1 = "";
+for (let i = hacker2.length; i--; ) {
+  result1 += hacker2[i];
+}
+console.log(result1);
+
+//8.
+if (hacker1[0] < hacker2[0]) {
+  console.log("The driver's name goes first");
+} else if (hacker1[0] > hacker2[0]) {
+  console.log("Yo, the navigator goes first definitely");
+} else {
+  console.log("What?! You both got the same name?");
+}
```

Comment:
Ok you by passed it a bit. here you are ask to compare the lexicographic order of the two strings. Lexicographic order is a bit wired but you can check the method: localCompare !